wgpu-types/src/lib.rs

```diff
@@ -837,46 +876,371 @@ pub enum TextureFormat {
+    Astc10x10RgbaUnorm = 84,
+    /// 10x10 block compressed texture. 16 bytes per block (1.28 bit/px). Complex pallet. 8 bit integer RGBA.
+    /// Srgb-color [0, 255] converted to/from linear-color float [0, 1] in shader.
+    ///
+    /// [`Features::TEXTURE_COMPRESSION_ASTC_LDR`] must be enabled to use this texture format.
+    Astc10x10RgbaUnormSrgb = 85,
+    /// 12x10 block compressed texture. 16 bytes per block (1.07 bit/px). Complex pallet. 8 bit integer RGBA.
+    /// [0, 255] converted to/from float [0, 1] in shader.
+    ///
+    /// [`Features::TEXTURE_COMPRESSION_ASTC_LDR`] must be enabled to use this texture format.
+    Astc12x10RgbaUnorm = 86,
+    /// 12x10 block compressed texture. 16 bytes per block (1.07 bit/px). Complex pallet. 8 bit integer RGBA.
+    /// Srgb-color [0, 255] converted to/from linear-color float [0, 1] in shader.
+    ///
+    /// [`Features::TEXTURE_COMPRESSION_ASTC_LDR`] must be enabled to use this texture format.
+    Astc12x10RgbaUnormSrgb = 87,
+    /// 12x12 block compressed texture. 16 bytes per block (0.89 bit/px). Complex pallet. 8 bit integer RGBA.
+    /// [0, 255] converted to/from float [0, 1] in shader.
+    ///
+    /// [`Features::TEXTURE_COMPRESSION_ASTC_LDR`] must be enabled to use this texture format.
+    Astc12x12RgbaUnorm = 88,
+    /// 12x12 block compressed texture. 16 bytes per block (0.89 bit/px). Complex pallet. 8 bit integer RGBA.
+    /// Srgb-color [0, 255] converted to/from linear-color float [0, 1] in shader.
+    ///
+    /// [`Features::TEXTURE_COMPRESSION_ASTC_LDR`] must be enabled to use this texture format.
+    Astc12x12RgbaUnormSrgb = 89,
+}
+
+macro_rules! texture_format_info_listing {
+    ($self:expr; $($name:ident => $feature:expr, $sample:expr, $srgb:expr, ($dim_x:literal, $dim_y:literal) in $size:literal),* $(,)?) => {
+        match $self {
+            $(
+                TextureFormat::$name => TextureFormatInfo {
+                    features: $feature,
+                    sample_type: $sample,
+                    block_dimensions: ($dim_x, $dim_y),
+                    block_size: $size,
+                    srgb: $srgb,
+                }
+            ),*
+        }
+    };
+}
+
+impl TextureFormat {
+    /// Get useful information about the texture format.
+    pub fn describe(&self) -> TextureFormatInfo {
+        #![allow(bindings_with_variant_name, non_snake_case)]
+
+        let NATIVE = Features::empty();
+        let BC = Features::TEXTURE_COMPRESSION_BC;
+        let ETC2 = Features::TEXTURE_COMPRESSION_ETC2;
+        let ASTC_LDR = Features::TEXTURE_COMPRESSION_ASTC_LDR;
+
+        let Uint = TextureSampleType::Uint;
+        let Sint = TextureSampleType::Sint;
+        let Nearest = TextureSampleType::Float { filterable: false };
+        let Float = TextureSampleType::Float { filterable: true };
+        let Depth = TextureSampleType::Depth;
+
+        let Linear = false;
+        let Srgb = true;
+
+        texture_format_info_listing! {
```

Comment:
do we actually need a macro here? it seems to me that we could just write it plain as:
```rust
use Self::*;
let (features, sample_type, block_dimensions, block_size, srgb) = match *self {
  R8Unorm => (NATIVE, Float, (1, 1), 1, Linear),
  ...
};
```

macros introduce another level of complexity, and they confuse both readers and tools (like `rustfmt`).

Reply:
Yeah, that's fine, the macro was needed before but overstayed its welcome.